shell/platform/windows/system_utils.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/system_utils.h"
+
+#include <Windows.h>
+
+#include <sstream>
+
+#include "flutter/shell/platform/windows/string_conversion.h"
+
+namespace flutter {
+
+std::vector<LanguageInfo> GetPreferredLanguageInfo() {
+  std::vector<std::wstring> languages = GetPreferredLanguages();
+  std::vector<LanguageInfo> language_info;
+  language_info.reserve(languages.size());
+
+  for (auto language : languages) {
+    language_info.push_back(ParseLanguageName(language));
+  }
+  return language_info;
+}
+
+std::vector<std::wstring> GetPreferredLanguages() {
+  std::vector<std::wstring> languages;
+  DWORD flags = MUI_LANGUAGE_NAME | MUI_UI_FALLBACK;
+
+  // Get buffer length.
+  ULONG count = 0;
+  ULONG buffer_size = 0;
+  if (!::GetThreadPreferredUILanguages(flags, &count, nullptr, &buffer_size)) {
```

Comment:
Since this is a win32 only API, I wonder if we should rename this file and it's header to be win32 specific.  The WinRT equivalent functionality is https://docs.microsoft.com/en-us/uwp/api/Windows.System.UserProfile.GlobalizationPreferences?redirectedfrom=MSDN&view=winrt-19041#Windows_System_UserProfile_GlobalizationPreferences_Languages and we'll have to abstract this to make language defaults work there, not sure we necessarily need to do that abstraction as part of this change but thinking we should be explicit about the win32 dependency

Reply:
I've renamed the implementation file to system_utils_win32.cc. There's nothing Win32-specific in the header, and for cases like that my preference is to have a unified header and differentiated implementation files (a pattern Chromium uses for a lot of its cross-platform code). I suspect some of our current win32_* files should be reconfigured that way as we add UWP support, although definitely not all of them.